File: Advanced Programming Methods/Toy Language C#/A2/Repository/IRepository.cs
using System;
using A2.UI;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A2.Repository
{
    public interface IRepository
    {
        void add(ProgramState s);
        ProgramState getCurrentProgram();
        void logProgramStateExec();
        void closeLogFile();
    }
}

File: Advanced Programming Methods/Toy Language Java/src/Repository/IRepository.java
package Repository;

import Model.DataStructures.MyException;
import UI.Menu.ProgramState;

import java.io.IOException;

public interface IRepository {
    void add(ProgramState s) throws MyException;
    ProgramState getCurrentProgram();
    void logProgramStateExec() throws IOException;
    void closeLogFile();
}
